### PR TITLE
Edit type/func documentation in the consumer package

### PR DESCRIPTION
**Pull request by eddyleelin**
**Description:** 
This PR updates the documentation of type and func declarations in the consumer package and all of its subpackages, with a focus on correcting grammar/spelling mistakes, cleaning up syntax to render the examples better, and updating incorrect references or wording. Note that this does not 

**Link to tracking Issue:** 
Addresses part of issue [#3050](https://github.com/open-telemetry/opentelemetry-collector/issues/3050).
Related to #3361 

**Documentation:**

- [x] Add number of upstream PR.

### CODE REVIEW

File: consumer/pdata/common.go [hunk 1]
```diff
@@ -65,12 +66,12 @@ func (avt AttributeValueType) String() string {
 //
 //   function f1(val AttributeValue) { val.SetIntVal(234) }
 //   function f2() {
-//   	v := NewAttributeValueString("a string")
-//      f1(v)
-//      _ := v.Type() // this will return AttributeValueTypeInt
+//       v := NewAttributeValueString("a string")
+//       f1(v)
+//       _ := v.Type() // this will return AttributeValueTypeInt
```
Comment (bogdandrutu) [inline consumer/pdata/common.go:71]: These are auto-generated from the cmd/pdatagen you need to change that code in order for these changes to persist.

Comment (eddyleelin) [reply consumer/pdata/common.go:71]: Thanks so much for letting me know that! I'll look into changing that.

**Comment by bogdandrutu** [comment]
Tests failed:

```
make build-binary-internal
make[1]: Entering directory '/home/circleci/project'
GO111MODULE=on CGO_ENABLED=0 go build -trimpath -o ./bin/otelcol_linux_amd64 -ldflags "-X go.opentelemetry.io/collector/internal/version.Version=v0.27.0-33-g84329591" ./cmd/otelcol
# go.opentelemetry.io/collector/consumer/pdata
consumer/pdata/generated_log.go:21:2: imported and not used: "go.opentelemetry.io/collector/internal/data/protogen/common/v1" as otlpcommon
consumer/pdata/generated_trace.go:21:2: imported and not used: "go.opentelemetry.io/collector/internal/data"
consumer/pdata/generated_trace.go:22:2: imported and not used: "go.opentelemetry.io/collector/internal/data/protogen/common/v1" as otlpcommon
make[1]: *** [Makefile:252: build-binary-internal] Error 2
make[1]: Leaving directory '/home/circleci/project'
make: *** [Makefile:163: otelcol] Error 2
```

**Review by bogdandrutu — COMMENTED** [review]
Can you please for me split this into 2 PRs:
1. cmd/pdatagen and consumer/pdata
2. Rest of the files

This will help me a lot for review.

**Comment by eddyleelin** [comment]
Thanks @bogdandrutu for the suggestion; I have updated this PR to not contain any of the cmd/pdatagen/internal or consumer/pdata/generated*.go files! Those files will be handled in the PR #3361 . Thanks again 😄

File: consumer/consumertest/err.go [hunk 2]
```diff
@@ -39,7 +39,7 @@ func (er *errConsumer) ConsumeLogs(context.Context, pdata.Logs) error {
 	return er.err
 }
 
-// NewErr returns a Consumer that just drops all received data and returns no error.
+// NewErr returns a Consumer that just drops all received data and holds the input error.
```
Comment (tigrannajaryan) [inline consumer/consumertest/err.go:42]:
```suggestion
// NewErr returns a Consumer that just drops all received data and returns the specified error to Consume* callers.
```

File: consumer/pdata/metric.go [hunk 3]
```diff
@@ -38,10 +38,10 @@ func (at AggregationTemporality) String() string {
 	return otlpmetrics.AggregationTemporality(at).String()
 }
 
-// Metrics is an opaque interface that allows transition to the new internal Metrics data, but also facilitate the
-// transition to the new components especially for traces.
+// Metrics is an opaque interface that allows transition to the new internal Metrics data, but also facilitates the
+// transition to the new components, especially for traces.
```
Comment (tigrannajaryan) [inline consumer/pdata/metric.go:42]: The reference to "traces" is confusing here. @bogdandrutu do you know what this is supposed to mean?

File: consumer/pdata/trace.go [hunk 4]
```diff
@@ -97,7 +97,7 @@ func (td Traces) ResourceSpans() ResourceSpansSlice {
 	return newResourceSpansSlice(&td.orig.ResourceSpans)
 }
 
-// TraceState in w3c-trace-context format: https://www.w3.org/TR/trace-context/#tracestate-header
+// TraceState is a message representing the tracestate in w3c-trace-context format: https://www.w3.org/TR/trace-context/#tracestate-header
```
Comment (tigrannajaryan) [inline consumer/pdata/trace.go:100]:
```suggestion
// TraceState is a string representing the tracestate in w3c-trace-context format: https://www.w3.org/TR/trace-context/#tracestate-header
```

**Review by tigrannajaryan — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
Please resolve the conflicts.

**Comment by alolita** [comment]
@bogdandrutu can you please review and this PR can then be merged if all is done.

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**